pygfx/utils/load.py

```diff
@@ -41,36 +41,60 @@ def load_meshes(path):
     ----------
     path : str
         The location where the mesh or scene is stored.
+        Can be a local file path or a URL.
 
     Returns
     -------
     meshes : list
         A list of loaded meshes.
 
     """
-
     import trimesh  # noqa
 
-    scene = trimesh.load(path)
+    # Trimesh's load() performs a similar check and refers
+    # loading from URLs to load_remote()
+    if "https://" in str(path) or "http://" in str(path):
+        scene = trimesh.load_remote(path)
+    else:
+        scene = trimesh.load(path)
+
+    # If this is a single object we can just convert and return it
     if isinstance(scene, trimesh.Trimesh):
         m = gfx.Mesh(
             gfx.geometry_from_trimesh(scene),
             gfx.MeshPhongMaterial(),
         )
         return [m]
-
+    # If this is a scene, we need to properly parse it
     elif isinstance(scene, trimesh.Scene):
+        # Scene consists of geometries (which in trimesh contain both
+        # the vertices/faces as well as the mateiral) and a scene graph.
+        # Each node in the graph consists of a geometry and a transform
+        # A geometry can be used by more than one node in the graph.
+        # Here we will convert each node in the graph to a pygfx.Mesh
+        # but in the future it might be better to use pygfx.InstancedMesh.
+        gfx_geometries = {}
+        gfx_materials = {}
+        meshes = []
         for node_name in scene.graph.nodes_geometry:
             transform, geometry_name = scene.graph[node_name]
-            current = scene.geometry[geometry_name]
-            current.apply_transform(transform)
-
-        return [
-            gfx.Mesh(
-                gfx.geometry_from_trimesh(m),
-                gfx.material_from_trimesh(m.visual.material),
-            )
-            for m in scene.geometry.values()
-        ]
+
+            # Convert each geometry only once
+            if geometry_name not in gfx_geometries:
+                m = scene.geometry[geometry_name]
+                gfx_geometries[geometry_name] = gfx.geometry_from_trimesh(m)
+                if hasattr(m.visual, "material"):
+                    gfx_materials[geometry_name] = gfx.material_from_trimesh(
+                        m.visual.material
+                    )
+                else:
+                    gfx_materials[geometry_name] = gfx.MeshStandardMaterial()
+
+            # Generate a visual for each node
+            mesh = gfx.Mesh(gfx_geometries[geometry_name], gfx_materials[geometry_name])
+            mesh.local.matrix = transform  # set the node's transform
+
+            meshes.append(mesh)
+        return meshes
```

Comment:
I see we are generating a flat list of meshes, but you mention a scene graph concept exists in trimesh, just like it does in pygfx. You _can_ replicate the actual tree in trimesh if you want to.

Reply:
True! I suppose that's what I was going for with this question:

> Is there appetite for functionality to load the entire trimesh scene, i.e. including lights, camera, background, etc?

I'll have a look into it but my thinking was that a `load_meshes` is expected to return meshes and nothing else. 

I could implement this in a separate function or via an additional argument (perhaps something like `return_scene=True`). Any preference?

Reply:
well. load_mesh was originally intended to load a single mesh. we already deviated from that quite a bit. personally I would expect a load function per filetype, but I guess in this case filetypes don't necessarily map to object types 1:1?